CmisSync.Lib/Sync/SynchronizedFolder.cs

```diff
@@ -1188,6 +1192,17 @@ private bool DownloadFile(IDocument remoteDocument, string localFolder)
                         if (null != remoteDocument.LastModificationDate)
                         {
                             File.SetLastWriteTime(filepath, (DateTime)remoteDocument.LastModificationDate);
+                        }
+
+                        // Should the local file be made read-only?

```

Comment:
yes. as i said in the comment: "set read only after we finish to manipulate the file or we can get an error if it's read only".
If you set a file read only and you try to modify it (eg SetLastModifiedDate) you'll get an error.
So we must set it read only when wi no longer need to manipulate it